src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,98 +164,296 @@ private void InitializeDefaultAutoCompleteCandidates()
+            }
+            catch(Exception e) {
                 DisplayNoRecommendationsLowConfidence = true;
-                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteLowConfidenceTitle;
-                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteLowConfidenceMessage;
-                LowConfidenceMessageAdditional = Resources.AutocompleteLowConfidenceMessageAditional;
-                DisplayLowConfidence = true;
+                DisplayLowConfidence = false;
+                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
+                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+                return;
+            }
+
+            var matchingResults = new List<NodeSearchElement>();
+            FilteredResults = new List<NodeSearchElementViewModel>();
+            var zeroTouchSearchElements = Model.SearchEntries.OfType<ZeroTouchSearchElement>().Where(x => x.IsVisibleInSearch);
+            var nodeModelSearchElements = Model.SearchEntries.OfType<NodeModelSearchElement>().Where(x => x.IsVisibleInSearch);
+
+            // no results
+            if (MLresults == null || MLresults.Results.Count() == 0)
+            {
+                DisplayNoRecommendationsLowConfidence = true;
+                DisplayLowConfidence = false;
+                NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
+                NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+                return;
+            }
+
+            // ML Results are categorized based on the threshold confidence score before displaying. 
+            if (MLresults.Results.Count() > 0)
+            {
+                FilteredHighConfidenceResults = new List<NodeSearchElementViewModel>();
+                FilteredLowConfidenceResults = new List<NodeSearchElementViewModel>();
+
+                foreach (var item in MLresults.Results)
+                {
+                    if (item.Node.Type.NodeType.Equals("FunctionNode"))
+                    {
+                        var element = zeroTouchSearchElements.FirstOrDefault(n => n.Descriptor.MangledName == item.Node.Type.Id);
+                        if (element != null)
+                        {
+                            element.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = item.Score;
+                            element.AutoCompletionNodeMachineLearningInfo.ViewConfidenceScoreRecentUse = true;
+                            element.AutoCompletionNodeMachineLearningInfo.IsByRecommendation = true;
+                            matchingResults.Add(element);
+                        }
+                    }
+                    else
+                    {
+                        // Retreive assembly name and full name from type id.
+                        var typeInfo = item.Node.Type.Id.Split(',');
```

Comment:
I think it would be good to use this constructor and avoid parsing it yourself.
https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname.-ctor?view=net-7.0#system-reflection-assemblyname-ctor(system-string)

Reply:
The type id we are sending has a different format than the Assembly path and was getting a exception while loading with that name format. I created a class to make the parsing robust.